src/v/cluster/controller_backend.cc

```diff
@@ -496,12 +496,39 @@ controller_backend::calculate_learner_initial_offset(
      * Initial learner start offset only makes sense for partitions with cloud
      * storage data
      */
+    if (auto topic_cfg = p->get_topic_config();
+        topic_cfg.has_value() && topic_cfg->get().is_internal()) {
+        vlog(clusterlog.trace, "{} is part of an internal topic", p->ntp());
+        return std::nullopt;
+    }
+
     if (!p->cloud_data_available()) {
         vlog(clusterlog.trace, "no cloud data available for: {}", p->ntp());
         return std::nullopt;
     }
 
+    if (p->get_cloud_storage_mode() != cluster::cloud_storage_mode::full) {
```

Comment:
do we still need the cloud_data_available()? (I think that that adds to confusion)

Reply:
not 100% sure, but this comment (which I don't really understand) gives me pause:

```
    // If the start offset for the STM region is not set, then the cloud log is
    // empty. There's one special case, where the start offset is set, and yet
    // the cloud log should be considered emtpy: retention in the STM region
    // advanced the start offset, but the garbage collection, and subsequent
    // truncation, did not happen yet.
```

Reply:
hmm ya, ok to skip then just to be on the safer side.